PeerioTechnologies/peerio-desktop
# WIP: room invite redesign

#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/273/desktop-room-invites-redesigned-flow
https://github.com/PeerioTechnologies/peerio-icebear/pull/141

#### Testing instructions  
This is what works so far:
* Room invites are listed along with chats, in alphabetical order, according to mockups
* Clicking room invite opens `ChannelInvite` component
* Host avatar is shown
* Accepting and rejecting invite works with existing SDK
  * very minor modifications to SDK so far; just made channel activate after accepting invite, which is how it used to work

SDK fixes (future):
* Need to be able to mark invites as `active` in SDK, the way rooms/DMs can be marked as `active`
  * Selecting a room invite does not currently set its entry in `ChatList` to `active` style, i.e. white text on blue background
  * ~~After declining an invite, selecting another room invite still shows the "invite declined" screen, because all room invites share the same component (will fix on client side for now)~~
* Similarly, need to be able to mark invites as `read`/`unread` in SDK
  * Selecting a room invite does not currently remove the `unread` style, i.e. semibold text
* No list of participants other than host

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
